### PR TITLE
Form stepper

### LINKED ISSUE #2: Setup application layout
Prepare layout that will be reused accross the app. It should help a user easily navigate in the app.

The layout consists of 2 parts:
- Navbar with a list of steps. UX of navbar:
    - A step is disabled unless its previous step is submitted.
    - A user can go to any available step
    - A user will be redirected to the last incompleted step if they try to go to the locked step
- Main part with forms.

### PR DESCRIPTION
**Pull request by coding-vibe**

### CODE REVIEW

File: src/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -1,9 +1,68 @@
...
+import StepLabel from '@mui/material/StepLabel';
+import FormScreens from 'constants/formScreens';
+import routes from 'constants/routes';
+
+const STEPS = Object.keys(FormScreens);
+
+function getActiveStep(path: string) {
+  switch (path) {
+    case '/personal-details':
+      return 0;
```
Comment (singvarr) [inline src/components/Layout/Layout.tsx]: Use constants for pathnames and put in constant ids of steps

File: src/contexts/WizardFormContext.tsx [hunk 2]
```diff
@@ -3,38 +3,51 @@ import FormScreens from 'constants/formScreens';
 
 export const initialFormValues = {
   [FormScreens.PERSONAL_DETAILS]: {
-    firstName: '',
-    lastName: '',
-    email: '',
-    phoneNumbers: [''],
+    order: 0,
```
Comment (singvarr) [inline src/contexts/WizardFormContext.tsx]: Move these digits to const

File: src/components/FullOrderDetails/FullOrderDetails.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ import WizardFormContext, {
 import * as classes from './styles';
 
 type InitialFormObjKeys = keyof InitialFormValuesType;
```
Comment (singvarr) [inline src/components/FullOrderDetails/FullOrderDetails.tsx]: Rename these types

File: src/components/FullOrderDetails/FullOrderDetails.tsx [hunk 4]
```diff
@@ -22,33 +22,43 @@ export default function FullOrderDetails() {
...
-                  sx={{ p: 1 }}>
-                  <span css={classes.fieldName}>{fieldName}:&nbsp;</span>
-                  <span css={classes.fieldValue}>{fieldValue}</span>
-                </Box>
-              ))}
-            </Paper>
-          ),
+      {Object.entries<InitialFormObjValues>(formValues).map(
+        ([screenName, screenInfo]) => {
+          if ('values' in screenInfo) {
```
Comment (singvarr) [inline src/components/FullOrderDetails/FullOrderDetails.tsx]: You don't need this thing. Filter by screen name

File: src/components/Layout/Layout.tsx [hunk 5]
```diff
@@ -1,8 +1,51 @@
...
+      case routes.DELIVERY_MODE:
+        return formValues.DELIVERY_MODE.order;
+      case routes.PAYMENT_METHOD:
+        return formValues.PAYMENT_METHOD.order;
+      case routes.FORM_SUBMISSION:
+        return formValues.FORM_SUBMISSION.order;
+      case routes.FORM_SUCCESS:
+        return formValues.FORM_SUCCESS.order;
+      default:
+        return formValues.PERSONAL_DETAILS.order;
```
Comment (singvarr) [inline src/components/Layout/Layout.tsx:29]: raise error here

File: src/components/WizardFormProvider/WizardFormProvider.tsx [hunk 6]
```diff
@@ -12,13 +15,92 @@ export default function WizardFormProvider({ children }: Props) {
...
+          ? {
+              subStep: {
+              id: FormScreens.COURIER_DELIVERY_DETAILS,
+                values: ???
+              },
+            }
+          : {
+              subStep: {
+                id: FormScreens.POST_DELIVERY_DETAILS,
+                values: ???
```
Comment (singvarr) [inline src/components/WizardFormProvider/WizardFormProvider.tsx]: Extract initial value (including `id` and `values`) for each substep and use here

File: src/components/WizardFormProvider/WizardFormProvider.tsx [hunk 7]
```diff
@@ -12,13 +15,92 @@ export default function WizardFormProvider({ children }: Props) {
...
+          ...update,
+        },
+      };
+    });
+  }, [formValues.DELIVERY_MODE.values]);
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    handleSaveFormValues((prevFormValues) => {
```
Comment (singvarr) [inline src/components/WizardFormProvider/WizardFormProvider.tsx:94]: Skip update at all if other payment method is selected

File: src/constants/stepOrder.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+enum ScreensOrder {
+  PERSONAL_DETAILS,
+  DELIVERY_MODE,
+  COURIER_DELIVERY_DETAILS,
+  POST_DELIVERY_DETAILS,
+  PAYMENT_METHOD,
+  CREDIT_CARD_DETAILS,
```
Comment (singvarr) [inline src/constants/stepOrder.ts:8]: Add TODO that substeps will be removed

File: src/contexts/WizardFormContext.tsx [hunk 9]
```diff
@@ -1,50 +1,53 @@
...
+    order: StepOrder.FORM_SUCCESS,
   },
 };
 
 export type InitialFormValuesType = typeof initialFormValues;
 
-export const saveFormValues = <T extends Exclude<FormScreens, 'FORM_SUCCESS'>>(
+export const saveFormValues = <
+  T extends Exclude<FormScreens, 'FORM_SUCCESS'>,
+  U extends Exclude<
```
Comment (singvarr) [inline src/contexts/WizardFormContext.tsx]: use `Pick` here

File: src/pages/CourierDeliveryDetailsScreen/CourierDeliveryDetailsScreen.tsx [hunk 10]
```diff
@@ -24,24 +21,37 @@ const MIN_DELIVERY_DATE = addDays(new Date(), MIN_DELIVERY_DAYS);
 const formatHandler = (value: number) => value || null;
 const parseHandler = (value: string) => value && parseInt(value, 10);
 
-type InitialValuesType =
-  InitialFormValuesType[FormScreens.COURIER_DELIVERY_DETAILS];
+type DeliveryDetailsType = {
```
Comment (singvarr) [inline src/pages/CourierDeliveryDetailsScreen/CourierDeliveryDetailsScreen.tsx]: use `typeof` on extracted substep instead. The same is applicable for other substeps

File: src/pages/DeliveryModeScreen/DeliveryModeScreen.tsx [hunk 11]
```diff
@@ -10,15 +10,16 @@ import WizardFormContext, {
...
+  InitialFormValuesType[FormScreens.DELIVERY_MODE]['values'];
 
 export default function DeliveryModeScreen() {
   const { onSaveFormValues } = useContext(WizardFormContext);
 
   return (
     <Form<DeliveryModeType>
       onSubmit={(values) => {
-        onSaveFormValues(FormScreens.DELIVERY_MODE, values);
+        onSaveFormValues(FormScreens.DELIVERY_MODE, values, null);
```
Comment (singvarr) [inline src/pages/DeliveryModeScreen/DeliveryModeScreen.tsx]: Don't forget that `null` is default value

File: src/components/Layout/Layout.tsx (hunk 5 above)
Comment (coding-vibe) [reply src/components/Layout/Layout.tsx:29]: we can only use numbers for activeStep prop in MUI Stepper

**Pull request closed without merging**